src/main/java/kz/toko/app/controller/errors/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,43 @@
+package kz.toko.app.controller.errors;
+
+import kz.toko.app.exception.EntityDeletedException;
+import kz.toko.app.exception.EntityNotFoundException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@Slf4j
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
```

Comment:
You don't need this annotation since you are already setting the response status in the `buildErrorResponse` method